crates/router/src/types/api/mandates.rs

```diff
@@ -106,7 +106,11 @@ impl MandateResponseExt for MandateResponse {
                 accepted_at: mandate.customer_accepted_at,
                 online: Some(api::payments::OnlineMandate {
                     ip_address: mandate.customer_ip_address,
-                    user_agent: mandate.customer_user_agent.unwrap_or_default(),
+                    // Using customer_user_agent as a fallback
+                    user_agent: mandate
+                        .customer_user_agent_extended
+                        .or(mandate.customer_user_agent)
```

Comment:
This can also be a method implemented on the type (without the `unwrap_or_else()` part).